crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -7,15 +7,140 @@ use ruma::{
 };
 use serde::{Deserialize, Serialize};
 
-/// The verification state of the device that sent an event to us.
+/// Message shield decoration color abstraction.
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub enum ShieldColor {
+    /// Trusted
+    GREEN,
+    /// Important warning
+    RED,
+    /// Low warning
+    GRAY,
+    /// No warning
+    NONE,
+}
+
+/// Directive for clients on how to decorate a decrypted message with this
+/// state.
 #[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ShieldState {
+    /// Recommended indicator color for this message
+    pub color: ShieldColor,
+    /// To display as a "tooltip" for users
+    pub message: Option<String>,
+}
+
+const AUTHENTICITY_NOT_GUARANTEED: &str =
+    "The authenticity of this encrypted message can't be guaranteed on this device.";
+const UNVERIFIED_DEVICE: &str = "Encrypted by an unverified device.";
+const UNSIGNED_DEVICE: &str = "Encrypted by a device not verified by its owner.";
```

Comment:
What's the difference between these two? Was this supposed to be `UNVERIFIED_IDENTITY` instead?